### DISCUSSION TITLE
Accessing filtered data on Data Table component

### DISCUSSION THREAD

**Discussion by B3nsten** [tusen-ai/naive-ui#3182] [Q&A]
Hello naive-Community,

is there any way to access the currently filtered items on the Data Table component?

For example, when using the "all" option on the selection column, the filter is ignored. The option will always select all items given by the data prop, no matter the filter state. This leads to unexpected results.

Only reference I could find is the itemCount variable on the pagination, which unfortunately is not very helpful to work with the data.

Can someone help me to get a better understanding of how to expose the data?

Thank you in advance.

**Comment by 07akioni** [comment]
Currently the feature is not provided. I think currently the only way is using `remote` prop and control all the data by yourself.

**Comment by 07akioni** [comment]
It would need further consideration if we should provide such data since it's a bit anti-pattern to single direction data flow.
